### ISSUE TITLE
jQuery not enqueued (after theme file modification)

### ISSUE THREAD

**Issue by foodlydoodlydoo** [infinum/wp-boilerplate#130]
## Current Behavior
I have a Wordpress plugin that depends on (use) jQuery.  I noticed jQuery was not enqueued by WP ('jQuery is not defined' js runtime errors).  Hence, I ran `npm install jQuery --save`.  After rebuild (kill and restart `npm start`) jQuery was enqueued.  But now when I modify a file (e.g. an .scss) after automatic rebuild jQuery no longer enqueues again.

A first-time build (after `npm start`) pages contain <script src="[snip]/mytheme/skin/public/jquery.min.js?ver=3.3.1">.  After file modification and rebuild the only difference in the reloaded markup is only that script tag missing.

I have not made any modifications to webpack.config.js from the default.

I'm new to npm and webpack, so this could well be just my misunderstanding of the configuration; there is not much doc for wp-boilerplate.

## Expected Behavior
Apparently, keep jQuery enqueued.  Looks like I should not even need to npm install jquery as webpack already provides it.

## Steps to Reproduce (for bugs)
See 'current behavior'

## Your Environment
* Development environment: Windows 10 x64 + subsystem for linux (ubuntu 18.04), npm 6.9.0, Visual Studio Code x64 (running npm from inside the IDE, internal terminal in VSCode changed to `bash` - works nicely)
* Server type: Apache x86 (local), working WP environment

**Comment by foodlydoodlydoo** [comment]
After the first build /skin/public/manifest.json contains "jquery.min.js": "//wp-content/themes/mytheme/skin/public/jquery.min.js".  After a file modification detection rebuild it no longer does (that line disappears).

**Comment by IvanGrginovInf** [comment]
Hi @foodlydoodlydoo, thanks for the detailed issue submission. It's a duplicate of Issue https://github.com/infinum/wp-boilerplate/issues/128 tho that one is in Croatian so no problem.

The issue should be fixed by the PR https://github.com/infinum/wp-boilerplate/pull/129 which will be merged in the next update.

Please feel more than free to test the fix on your setup.

**Comment by foodlydoodlydoo** [comment]
Confirming that locally applied https://github.com/infinum/wp-boilerplate/pull/129/commits/bb0090345b3e0f8fc87e8159ad90462415391058 fixes the issue for me.  Thanks.  Duplicate this to #129 at will.

A novice question: after #129 is merged to production, how do I update?  git pull + merge?  some npm magic?  Thanks!

**Comment by IvanGrginovInf** [comment]
That's great to hear, thank you for testing and reporting back!

As for updating, we envisioned this more as being ejected from the start for a couple of reasons. 

1. This is a boilerplate, not a theme. You're supposed to modify it almost entirely and build your own theme for which **you** are in charge of keeping it up to date.
2. In the process of installation we do multiple things to customise / whitelabel this for your own future theme (such as renaming namespaces in all files). This process makes most of your local boilerplate files different from the github from the very start.
3. To add to that, in the process of building your own theme, you will definitely modify many core files making updating them really tricky (this doesn't / can't work as a child theme where we update only the files you haven't touched - mostly due to the namespaces renaming)

That being said, I guess you **_could_** `git pull` and **selectively** merge **some** changes. BUT. You have to be **extra** careful when doing this and only merge **specific changes** which you absolutely need (not whole files). Just straight up merging an enitre file (or more of them) will most likely break something.

Hope that's answers your question even tho it might not be answer you wanted. 😄

**Comment by foodlydoodlydoo** [comment]
Thank you!  Makes perfect sense - a bit like a motherboard BIOS update - don't do it, unless you really need to :)  

Cherry picking fixes when something broken is found is probably the way.  Definitely subscribing to changes in this repo is not a bad idea.

Thanks again.  I'll close this now in favor of #129.
